### ISSUE TITLE
Sounds never work - relative path error

### ISSUE THREAD

**Issue by Ramadel** [coffiarts/FoundryVTT-crunch-my-party#7]
I constantly get this error with default settings...

![image](https://github.com/coffiarts/FoundryVTT-crunch-my-party/assets/54110249/4a5ed22f-a6b3-410b-b777-36a7c595a13a)

If you uncheck the box to not play the sound...

![image](https://github.com/coffiarts/FoundryVTT-crunch-my-party/assets/54110249/cf0d1e04-e34f-4940-9c31-5007019bc567)

I wish I could just give it an absolute path. The files are there.

**Comment by coffiarts** [comment]
Am I assuming correctly that your mod settings look like shown below, and that you've used the **file picker button** for assigning your audio files?
Because that's how it is meant to be and how it should work since a couple of mod updates:
![image](https://github.com/coffiarts/FoundryVTT-crunch-my-party/assets/39616483/aa9d21b3-8f19-4715-9ee0-d3de01308e4a)

**Comment by Ramadel** [comment]
Yes, that is exactly what it looks like.

**Comment by coffiarts** [comment]
Hmmmm... but this is exactly how it works for me - both remote (on Forge) and in a local session running on electron.
Two more questions to narrow it down:
1. Does it occur in the GM oder player session (or in both)?
2. Did you change/replace the audio files that came bundled with my mod?

**Comment by Ramadel** [comment]
1. It happens in both, though (either in Sequencer or CMP, can't remember) I can suppress it for the players.
2. Nope. Never touched it. The sound files have actually never worked and I've always received this message. I'm new to the mod, but I love it! (Minus this annoying error popup.)

**Comment by coffiarts** [comment]
Well, that's tricky then... having a hard time reproducing this. But I understand that it's very ugly that way, so I really hope that I can resolve it. Let's see...

Could you please enter the following commands into your console and post me the results?

```
game.settings.get('crunch-my-party', 'playAudio4Crunch')
```
```
game.settings.get('crunch-my-party', 'audioFile4Crunch')
```
```
game.settings.get('crunch-my-party', 'playAudio4Explode')
```
```
game.settings.get('crunch-my-party', 'audioFile4Explode')
```
This is what it returns on my side (everything enabled, with default audio files:
![image](https://github.com/user-attachments/assets/82492b94-83bc-4648-b139-3d077d2087b6)

Furthermore, could you turn **debugging** on for _both_ CMP _and_ Sequencer like this:
![image](https://github.com/user-attachments/assets/49a002c3-9176-4d0b-bb28-34ed442ff390)
![image](https://github.com/user-attachments/assets/9ceb843d-1b45-4e45-9a54-24bc5c9294f1)

... and then **reproduce the error** and send me your **log output**?

On my side, it looks like this:
![image](https://github.com/user-attachments/assets/d696e74d-81b8-46dd-bc7c-b4a3d600ce65)

Hoping that this will get me some leads.

**Comment by Ramadel** [comment]
![image](https://github.com/user-attachments/assets/116ccc06-03ed-4a25-bad0-95678ac3d402)

Debugging results
![image](https://github.com/user-attachments/assets/1b9e410f-cc9f-4a0f-a35c-e1a1a66976e6)

**Comment by coffiarts** [comment]
Thx. Looking into it...

**Comment by coffiarts** [comment]
I confess that this is really weird. Somehow your setup seems to use another base dir than mine at runtime while accessing the audio file.
I see that I am talking to someone capable of going into technical details here - so my suggestion (and only idea) to track this down is to hack my code a little bit on your side, so that it logs out the base path it is using... given that don't mind dive so deeply into the depths.

Could you please edit Data/modules/crunch-my-party/scripts/main.js by adding the following line..
```javascript
Logger.debug(`Audio base dir (window.location.pathname): ${window.location.pathname}`);
```
... twice, i.e. once after line 655 and once after line 744, like this:

![image](https://github.com/user-attachments/assets/c8e557ab-5614-4d20-92bd-d53c8450d239)

![image](https://github.com/user-attachments/assets/f124a412-839e-42dc-93aa-d23f4485f7df)

After that, filter your console to "audio base dir" and reload your current scene. Capture the console output.

Mine looks like this:
![image](https://github.com/user-attachments/assets/d6f38d91-2ab8-4b96-ae65-f35a92bc2c99)

In addition, although I absolutely believe you when you say the files _are_ there: Could you still post me a screenshot of your file structure where they are residing. One can never know for sure...

Sorry for so much debugging, and thanks for your patience. This is really a tough nut!

**Comment by coffiarts** [comment]
@Ramadel - Please forget about my last post (about the hacking) - maybe there's an easier way:

Please go to https://github.com/coffiarts/FoundryVTT-crunch-my-party/blob/%237-fix/src/crunch-my-party/scripts/main.js
It is an updated script with an experimental fix.
Download this file (main.js), oder copy & paste its content to your file at Data/modules/crunch-my-party/scripts/main.js, overwriting everything.
Then reload your scene and try again.

I am curious to know if it works.

**Comment by Ramadel** [comment]
I did your last post. Same result. I made a backup of the original main.js first.

**Comment by coffiarts** [comment]
Seems like I am running out of ideas here.

**Two more questions:**
1.) Does the same error occur when running your game locally on Foundry's electron client?
2.) Do you have the chance to run it through another server instance, e.g. on Forge?

**And an extended debugging option:**
Could you please download/replace your main.js once more from https://github.com/coffiarts/FoundryVTT-crunch-my-party/blob/%237-fix/src/crunch-my-party/scripts/main.js and try again?
It does not fix it, but adds more debugging. Please look out for the following lines in your console and post here what it looks like on your end:
![image](https://github.com/user-attachments/assets/26c8207b-aa3e-4e96-a0e2-e206a7545194)

**Comment by Ramadel** [comment]
![image](https://github.com/user-attachments/assets/dd13d7f2-edf2-4170-aac1-d4ea564d0ca3)

![image](https://github.com/user-attachments/assets/f1538938-9ccf-484f-8ee5-182ee01ea127)

/foundry/game is the difference. That probably explains it. This is a default install.

**Comment by coffiarts** [comment]
Alright, guess we're almost there!
Could you please try again with the change I uploaded a sec ago?
https://github.com/coffiarts/FoundryVTT-crunch-my-party/blob/%237-fix/src/crunch-my-party/scripts/main.js

**Comment by Ramadel** [comment]
That did it!

**Comment by coffiarts** [comment]
Hurray! Glad to hear it.
The interesting part is that this marginal fix of the relative paths also work in my environment. So I assume that my original paths were wrong, and that only your specific installation, as stumbling block, has revealed this. Why they have been working all the time for me and others remains a mystery for me. But I believe that I can roll this out as regular solution for everyone.
This can take a couple of days, so I recommend that you do not reinstall my mod during that time, but keep to the patch. Once the new version is out, you can then safely return to a regular install.
Thanks alot for finding and testing this so patiently!

**Comment by coffiarts** [comment]
Fixed by latest release [12.0.1](https://github.com/coffiarts/FoundryVTT-crunch-my-party/releases/tag/latest)
